### PR TITLE
Make test container names informative

### PR DESCRIPTION
**Pull request by nick-drozd**
Here's a typical test error message, such as you might see posted in Slack:
```
integration_test_1  | ======================================================================
integration_test_1  | FAIL: test_intkey_smoke (test_intkey_smoke.TestIntkeySmoke)
integration_test_1  | ----------------------------------------------------------------------
integration_test_1  | Traceback (most recent call last):
integration_test_1  |   File "/project/sawtooth-core/integration/sawtooth_integration/tests/test_intkey_smoke.py", line 84, in test_intkey_smoke
integration_test_1  |     self.post_and_verify(batch, how_many_updates)
integration_test_1  |   File "/project/sawtooth-core/integration/sawtooth_integration/tests/test_intkey_smoke.py", line 95, in post_and_verify
integration_test_1  |     self.verify_state_after_n_updates(how_many_updates)
integration_test_1  |   File "/project/sawtooth-core/integration/sawtooth_integration/tests/test_intkey_smoke.py", line 105, in verify_state_after_n_updates
integration_test_1  |     'Updated state error')
integration_test_1  | AssertionError: {'thrush': 325, 'finch': 651, 'jay': 538, 'lark': 415, 'wren': 437} != {}
integration_test_1  | - {'finch': 651, 'jay': 538, 'lark': 415, 'thrush': 325, 'wren': 437}
integration_test_1  | + {} : Updated state error
```
Which test failed? Apparently an intkey test, but which one? Python? Java? _Go_? Who knows? Currently the only way to tell is to dig through the docker output, which is very unpleasant.

This PR renames the test containers (that is, the docker containers that actually run the tests and report results) to match the names of their docker-compose files. It also does away with the `run_docker_test` `-s` flag, which makes it easier to use.

Here's what the test output looks like with these changes:
```
intkey-java-smoke_1  | ======================================================================
intkey-java-smoke_1  | FAIL: test_intkey_smoke (test_intkey_smoke.TestIntkeySmoke)
intkey-java-smoke_1  | ----------------------------------------------------------------------
intkey-java-smoke_1  | Traceback (most recent call last):
intkey-java-smoke_1  |   File "/project/sawtooth-core/integration/sawtooth_integration/tests/test_intkey_smoke.py", line 84, in test_intkey_smoke
intkey-java-smoke_1  |     self.post_and_verify(batch, how_many_updates)
intkey-java-smoke_1  |   File "/project/sawtooth-core/integration/sawtooth_integration/tests/test_intkey_smoke.py", line 95, in post_and_verify
intkey-java-smoke_1  |     self.verify_state_after_n_updates(how_many_updates)
intkey-java-smoke_1  |   File "/project/sawtooth-core/integration/sawtooth_integration/tests/test_intkey_smoke.py", line 105, in verify_state_after_n_updates
intkey-java-smoke_1  |     'Updated state error')
intkey-java-smoke_1  | AssertionError: {'jay': 538, 'finch': 651, 'thrush': 325, 'wren': 437, 'lark': 415} != {}
intkey-java-smoke_1  | - {'finch': 651, 'jay': 538, 'lark': 415, 'thrush': 325, 'wren': 437}
intkey-java-smoke_1  | + {} : Updated state error
```

### CODE REVIEW

**Comment by nick-drozd** [comment]
Hey look, my build failed. Let's see which test it was:
```
dynamic-network_1  | [20:54:15.157 ERROR    cli] Address already in use
dynamic-network_1  | Traceback (most recent call last):
dynamic-network_1  |   File "/project/sawtooth-core/validator/sawtooth_validator/server/cli.py", line 283, in main
dynamic-network_1  |     validator.start()
dynamic-network_1  |   File "/project/sawtooth-core/validator/sawtooth_validator/server/core.py", line 492, in start
dynamic-network_1  |     self._service.start()
dynamic-network_1  |   File "/project/sawtooth-core/validator/sawtooth_validator/networking/interconnect.py", line 597, in start
dynamic-network_1  |     raise err
dynamic-network_1  |   File "/usr/lib/python3.5/threading.py", line 914, in _bootstrap_inner
dynamic-network_1  |     self.run()
dynamic-network_1  |   File "/usr/lib/python3.5/threading.py", line 862, in run
dynamic-network_1  |     self._target(*self._args, **self._kwargs)
dynamic-network_1  |   File "/project/sawtooth-core/validator/sawtooth_validator/networking/interconnect.py", line 348, in setup
dynamic-network_1  |     self._socket.bind(self._address)
dynamic-network_1  |   File "zmq/backend/cython/socket.pyx", line 487, in zmq.backend.cython.socket.Socket.bind (zmq/backend/cython/socket.c:5156)
dynamic-network_1  |   File "zmq/backend/cython/checkrc.pxd", line 25, in zmq.backend.cython.checkrc._check_rc (zmq/backend/cython/socket.c:7535)
dynamic-network_1  | zmq.error.ZMQError: Address already in use
dynamic-network_1  | Exception in thread InterconnectThread-5:
dynamic-network_1  | Traceback (most recent call last):
dynamic-network_1  |   File "/usr/lib/python3.5/threading.py", line 914, in _bootstrap_inner
dynamic-network_1  |     self.run()
dynamic-network_1  |   File "/usr/lib/python3.5/threading.py", line 862, in run
dynamic-network_1  |     self._target(*self._args, **self._kwargs)
dynamic-network_1  |   File "/project/sawtooth-core/validator/sawtooth_validator/networking/interconnect.py", line 348, in setup
dynamic-network_1  |     self._socket.bind(self._address)
dynamic-network_1  |   File "zmq/backend/cython/socket.pyx", line 487, in zmq.backend.cython.socket.Socket.bind (zmq/backend/cython/socket.c:5156)
dynamic-network_1  |   File "zmq/backend/cython/checkrc.pxd", line 25, in zmq.backend.cython.checkrc._check_rc (zmq/backend/cython/socket.c:7535)
dynamic-network_1  | zmq.error.ZMQError: Address already in use
```

File: consensus/poet/cli/tests/unit_cli.yaml [hunk 1]
```diff
@@ -17,7 +17,7 @@ version: "2.1"
 
 services:
 
-  poet:
+  unit_cli:
```
Comment (aludvik) [inline consensus/poet/cli/tests/unit_cli.yaml]: Change this and the file name to unit_poet_cli

File: consensus/poet/families/tests/tp-validator-registry.yaml [hunk 2]
```diff
@@ -23,14 +23,14 @@ services:
...
-    command: tp_validator_registry -vv tcp://validator:40000
+    command: tp_validator_registry -vv tcp://tp-validator-registry:40000
     stop_signal: SIGKILL
     environment:
         PYTHONPATH: "/project/sawtooth-core/consensus/poet/families/:\
             /project/sawtooth-core/consensus/poet/common:\
             /project/sawtooth-core/sdk/python"
 
-  validator:
+  tp-validator-registry:
```
Comment (aludvik) [inline consensus/poet/families/tests/tp-validator-registry.yaml]: Using - and _ to differentiate service names is confusing

**Comment by aludvik** [comment]
Couple changes, otherwise looks good.

File: bin/run_tests [hunk 3]
```diff
@@ -193,133 +193,134 @@ main() {
...
+    copy_coverage .coverage.test_poet_smoke
+    run_docker_test test_dynamic_network --timeout 600
+    copy_coverage .coverage.test_dynamic_network
+    run_docker_test test_basic_auth_proxy
+    copy_coverage .coverage.test_basic_auth_proxy
 }
 
 test_deployment() {
-    run_docker_test systemd_services -s integration_test
+    run_docker_test systemd_services
```
Comment (jsmitchell) [inline bin/run_tests]: I think this needs to be test_systemd_services

**Pull request merged**